evalml/pipelines/components/transformers/preprocessing/time_series.py

```diff
@@ -0,0 +1,57 @@
+import pandas as pd
+
+from evalml.pipelines.components.transformers.transformer import Transformer
+
+
+class DelayedFeaturesTransformer(Transformer):
+    name = "Delayed Features Transformer"
+    hyperparameter_ranges = {}
+    needs_fitting = False
+
+    def __init__(self, max_delay=2, random_state=0, **kwargs):
```

Comment:
@freddyaboulton can we add parameters to control delaying the target vs delaying the features?
```python
delay_target=False,
delay_features=True
```
I think it'll be helpful to be able to control the behavior like this.

Reply:
@dsherry Should these be tuned by automl?

Reply:
@freddyaboulton good question. Thinking about that, I think we should:
* Default these both to True in this PR
* Once we get start performance testing for timeseries we can determine whether or not we should make these tuneable. But my default assumption would be to leave it as non-tuneable

We need the ability to generate both delayed features and delayed target features. But for many datasets, `target[n-1]` will be strongly correlated with `target[n]`, and users may not want their models to rely so heavily on the delayed target, and will want to disable the delayed target features. Hence why I suggested we add these two parameters.

Reply:
Great point about the previous time step being correlated with the current time step! 

I agree with your suggestion to leave the question about tuning until we have performance tests. I was thinking about the case when there are many delayed features. In that situation, not all of them will be useful for modeling the target and so we may want to intelligently not delay some of those features. There are other, and probably better ways of parametrizing that though, apart from the coarse "should we delay or not delay" decision.

Reply:
@freddyaboulton agreed. These two binary parameters are not going to be enough for us for tuning in the long-term.